source/Projectile.cpp

```diff
@@ -90,7 +90,7 @@ Projectile::Projectile(const Projectile &parent, const Weapon *weapon)
 
 
 // Ship explosion.
-Projectile::Projectile(Point position, const Weapon *weapon)
+Projectile::Projectile(const Point &position, const Weapon *weapon)
 	: weapon(weapon)
 {
 	this->position = position;
```

Comment:
The "better" change would be to `std::move` the by-value parameter here, to use move assignment rather than copy assignment.

Reply:
Doesn't the compiler automatically use the move-assignment when it detects that the given parameter will never be used again? (If that is the case, then I'll just put back the original code since that is more readable.)

Reply:
It is forbidden to automatically move from an "lvalue", which is effectively any variable with a local name.
The compiler can optimize to elide _copies_ (e.g. return value optimization) in a variety of cases, but assignment isn't one of them.

So when one has a local variable and basically the whole point is to put it into something else (and never use that local variable again _without first assigning back to it_), we can signal that move ops are allowable via `std::move`. There can be scenarios where this inhibits optimizations (like RVO), and in many of them a compiler warning is generated.

Reply:
Thanks, very clear! Then adding `std::move` is indeed a good change; I added it.